src/HealthCheck.Core/Results/HealthCheckResult.cs

```diff
@@ -0,0 +1,22 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+
+namespace HealthCheck.Core.Results
+{
+    public class HealthCheckResult
+    {
+        private readonly ICheckResult[] _results;
+        private readonly bool _passed;
+
+        public ICheckResult[] Results => _results;
+        public bool Passed => _passed;
+        public string Output => _passed ? "SUCCESS" : "FAILURE";
+
+        public HealthCheckResult(IEnumerable<ICheckResult> results)
+        {
+            _results = results.OrderBy(x => x.CheckerName).ToArray();
+            _passed = _results.Any() && _results.All(x => x.Passed);
```

Comment:
I don't see the value in inverting the logic of empty healthchecks (no checkers). A health check passes if there are no failures, there should be no requirement for "at least one checker" - alternatively the `HealthCheck` constructor should throw if no checkers are passed to it.